File: 20. Valid Parentheses/Solution.cpp
class Solution
{
  public:
    bool isValid(string s)
    {
        vector<char> stack;
        for (string::iterator i = s.begin(); i != s.end(); i++)
        {
            if (*i == '{' || *i == '[' || *i == '(')
                stack.push_back(*i);
            else
            {
                if (stack.empty())
                    return false;
                char c = stack[stack.size() - 1];
                if ((c == '(' && *i == ')') || (c == '[' && *i == ']') || (c == '{' && *i == '}'))
                    stack.pop_back();
                else
                    return false;
            }
        }
        return stack.empty() ? true : false;
    }
};

File: 20. Valid Parentheses/Solution.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        for i in s:
            if i in ['{', '(', "["]:
                stack.append(i)
            else:
                if not stack:
                    return False
                c = stack[-1]
                if (c == '(' and i == ')') or (c == '['
                                               and i == ']') or (c == '{'
                                                                 and i == '}'):
                    stack.pop()
                else:
                    return False
        return True if not stack else False